### ISSUE TITLE
Visibility computations by client rectangles, vertical writing mode (column layout)

### ISSUE THREAD

**Issue by danielweck** [readium/readium-shared-js#87]
@yuriyp 
@idov-isd 

Yuri, Igor?

Given your past contributions to the "rectangle" algorithms in CfiNavigationLogic, any chance you could help me figure-out how to implement support for vertical writing mode + right-to-left text/page progression?

https://github.com/readium/readium-shared-js/pull/16

I would really appreciate your help, as this is a blocking issue that prevents supports for Japanese EPUBs:
not just Media Overlays, but also linking (TOC), bookmarks, etc. (anything that requires mapping an element to a page/column number)

I started a feature branch:

https://github.com/readium/readium-shared-js/tree/feature/verticalColumns

...and as you can see I did not get very far:

https://github.com/readium/readium-shared-js/commit/c6a52be291b871eeac12738175c530eb8d35139b

Test EPUB (chapter 1,2,3 with Media Overlays):

kusamakura-japanese-vertical-writing-20121124
https://code.google.com/p/epub-samples/downloads/detail?name=kusamakura-japanese-vertical-writing-20121124.epub

In a nutshell: the CSS columns width is actually their height, because of the vertical flow (they are visually stacked from top to bottom). The text goes from right to left, and flows in thin top-to-bottom columns. There is no two-page spread, only a single column per visible viewport.

Many thanks!
Regards, Daniel
